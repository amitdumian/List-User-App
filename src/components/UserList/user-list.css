
.userImage{
    border-radius:50%;
    position: absolute;
    left:38%;
    top: -35px;
}

.flex-row-container {
    padding: 10%;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    width: 85%;
}

 .flex-row-item {
    width: 29%;
    border-radius: 10px;
    margin-bottom: 5%; 
    position: relative;
    background: ghostwhite; 
    border:1px solid #fff; 
    height: 160px;
 }

.flex-row-item:hover{
    background-color:lightblue;
    cursor: pointer;
 }

 .pipe{
     font-weight: 200;
 }

.user-info{
    padding: 20%;
    /*white-space: nowrap;*/
}

.user-name{
    font-weight: bold;
    font-size: x-small;
}

.user-location{
    font-size: x-small;
}

.learn-more-div{
   padding-top:5px;
}

.learn-more{
    font-size: x-small;
    color: deepskyblue;
    padding-left: 40%;
    font-weight: 600;
    cursor: pointer;
    text-decoration: none;
}




