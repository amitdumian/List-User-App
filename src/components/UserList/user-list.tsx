import { useState, useEffect } from "react";
import  "./user-list.css";
import UserDetailsTemplate from "../UserDetails/user-details-template";
import ReactModal from "../Common/react-modal";

const UserList = () => {  
    const [userList, setUserList] = useState([]);
    const [isLoading, setIsLoading] = useState(true);
    const [showModal, setShowModal] = useState(false);
    const [userDetails, setUserDetails] = useState();
    const apiUrl = 'https://randomuser.me/api/';
    const loadingMsg = 'Please Wait for Users to Load';
    const userCount = 9;
    const learnMoreText = "Learn More";
  
    const toggleModal =() => {
      setShowModal(!showModal);
    }

    useEffect(() => {
        fetch(apiUrl)
          .then(res => res.json())
          .then(response => {
            const arrUser:any = [];
            for(let i = 0; i < userCount; i++){
              arrUser.push(response.results[0]);
            }
            setUserList(arrUser);
            setIsLoading(false);
          })
          .catch(error => console.log(error));
      }, []);

      return (
        <div className={"flex-row-container"}> 
        {isLoading && <p>{loadingMsg}</p>}
        {userList && userList.map((user:any, index) => (
          <>
            <div className={"flex-row-item"} key={index}>
                <>
                  <div className = {"user-info"}>
                    <span className = {"user-name"}>{user.name.first + ' ' + user.name.last}</span>
                    <span className = {"pipe"}>{` | `}</span>
                    <span className = {"user-location"}>{user.location.state + ", " + "United States"}</span>
                    <span><img src={user.picture.medium} className={"userImage"}/></span>
                    <div className = {"learn-more-div"}>
                      <a 
                        className = {"learn-more"}
                        onClick={()=> {
                          setUserDetails(user);
                          setShowModal(!showModal);
                        }
                        }>
                        {learnMoreText}</a>
                    </div>
                  </div>
                </>
            </div>
            {showModal && userDetails && 
            <ReactModal show={showModal} handleClose={toggleModal}>
                <UserDetailsTemplate userDetails = {userDetails}></UserDetailsTemplate>
            </ReactModal>}
           </>
        ))}
      
        </div>
  
    );
      
}

export default UserList;
