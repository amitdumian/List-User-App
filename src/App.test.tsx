
import App from './App';
import { shallow } from 'enzyme';
import UserList from './components/UserList/user-list';
import renderer from 'react-test-renderer';

describe('Parent Component', () => {
  it('renders Child component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<UserList />)).toEqual(true);
  });

  it('should render userlist on load', async () => {
    const elem = renderer.create(<App />).toJSON();
    expect(elem).toMatchSnapshot();
 });
});

