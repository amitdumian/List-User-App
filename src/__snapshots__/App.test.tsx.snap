// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent Component should render userlist on load 1`] = `
<div>
  <div
    className="flex-row-container"
  >
    <p>
      Please Wait for Users to Load
    </p>
  </div>
</div>
`;
